internal/commands/push/inputs.go

```diff
@@ -47,17 +49,20 @@ func (i *inputs) Resolve(profile *user.Profile, ui terminal.UI) error {
 		searchPath = profile.WorkingDirectory
 	}
 
+	_, err := os.Stat(path.Join(profile.WorkingDirectory, i.LocalPath))
```

Comment:
```suggestion
	_, err := os.Stat(searchPath)
```

Reply:
this might have to stay as-is since the `os.Stat` fails to find the path if searchPath is just the "local path"

Reply:
ah yeah I think I see what I was missing before...

so my question to you is: if/when the `LocalPath` does not exist, does `FindApp` return an error?  or an empty `App`?

if it returns an error, maybe we need to look for that specific error in the `err != nil` case of `local.LoadAppConfig`.  otherwise, I think instead of just my suggestion here, you may need to call `filepath.Abs(searchPath)` first, before doing `os.Stat`

Reply:
`FindApp` does not return an error when `LocalPath` does not exist. It just returns an empty `App`.

I added the filepath.abs check for `path.Join(profile.WorkingDirectory, i.LocalPath)` instead of just `searchPath`
otherwise it would fail to find the path as such: 
```
=== RUN   TestPushInputsResolve/should_return_an_error_when_specified_local_path_is_not_a_realm_app_project
    inputs_test.go:57: 
        failed to assert equals ( actual, expected )
                push.errProjectInvalid{directory 'pathWithNoApp' does not exist}
                push.errProjectInvalid{directory 'pathWithNoApp' is not a supported Realm app project}
```

Reply:
the issue with relying on `path.Join(profile.WorkingDirectory, i.LocalPath)` is that when `--local` is the fully qualified path to an app directory, this will fail (because it will prepend `WorkingDirectory` to the fully qualified path).

ultimately, what we're looking for is:

* when `--local` is not set, we assume we're looking in the current working directory
* when `--local` is set to a relative directory (basically when it doesn't begin with either `/` or `~`), we assume we're looking in the current working directory PLUS that local path
* when `--local` is set to a fully qualified path, we assume we're just looking there

thankfully, using `filepath.Abs` on what we define as `searchPath` handles most of this complexity (that's why the `local.FindApp` function works the way it does)

I think after you determine `searchPath`, instead of calling `filepath.Abs` on the joined paths like you have, you instead want:

```go
// we'll replace searchPath with its "absolute representation"
searchPath, err = filepath.Abs(searchPath)
if err != nil {
	return err
}

// you can os.Stat(searchPath) now and see if it even exists first

// its this searchPath that we should ultimately pass to our error for messaging
```